### PR TITLE
DOC: show gains realized by sampling more queries

### LINKED ISSUE #62: Better compare NEXT and Salmon
Currently, there's a beta comparison of NEXT and Salmon on the documentation: https://docs.stsievert.com/salmon/benchmarks/adaptive.html

It'd be nice to have a more firm benchmark, perhaps one with the "strange fruit" where the ground truth is known (which also allows for easier simulation).

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
It shows the benefit of searching more queries, both in the documentation and as an example. Here's the complete list of what this PR does:

- [x] Provides an example offline usage, which shows the difference of number of queries to search.
- [x] Implements and tests `score` and `predict` for offline usage.
- [x] Creates a `dataset.py` and `datasets` directory in `examples`. This enables usage of these datasets in other (future) examples.
- [x] Adds comparison to documentation

In implementing the comparison, I've also made these fixes:

* Passed `random_state` correctly when getting random queries
* Removed `initial_batch_size` from `Embedding`; now it (almost) runs gradient descent. `initial_batch_size` is only used in the `Damper` children.
* Add learning rate damper
* Add dataset loading utility to the examples. This utility includes the strange-fruit and Zappos dataset.
* Return exactly the number of queries requested.
* Make sure the query scoring doesn't raise any FloatingPointErrors (which are actually warnings).

**Reference issues/PRs**
* #62 is relevant because Salmon searches far more queries than NEXT. I think this PR will close #62.

### CODE REVIEW

**Comment by stsievert** [comment]
I've run some simulations on the strange fruit dataset. This is a synthetic use case that uses a response model derived from real human responses. I plan on integrating this result (or a very similar result) into the documentation:

Screen Shot 2020-08-31 at 2 18 45 PM

There are still some bugs I need to resolve with this.
